httpserver/server.go

```diff
@@ -192,3 +189,34 @@ type ServerAddrNotFoundError struct{}
 func (err ServerAddrNotFoundError) Error() string {
 	return "The http server address configuration is not set. Skipping server creation"
 }
+
+func serverGracefulShutdown(isTLSEnabled bool, svr *http.Server, lsnr net.Listener, certFile string, keyFile string) error {
+	connectionsClosed := make(chan struct{})
+	go func() {
+		sigint := make(chan os.Signal, 1)
+		signal.Notify(sigint, os.Interrupt, syscall.SIGTERM)
+		<-sigint
+		logrus.Info("shutting down ratify server...")
+		ctx, cancel := context.WithTimeout(context.Background(), 6*time.Second) // wait 6 seconds for existing connections to finish
+		defer cancel()
+		if err := svr.Shutdown(ctx); err != nil {
+			logrus.Errorf("failed to shutdown ratify server: %v", err)
+		}
+		close(connectionsClosed)
+	}()
+	if isTLSEnabled {
+		if err := svr.ServeTLS(lsnr, certFile, keyFile); err != nil && err != http.ErrServerClosed { // http.ErrServerClosed is immediately returned when server.Shutdown() is called
```

Comment:
not a strong opinion, wonder if we want to move tls cert watcher under here?

Reply:
yeah definitely possible but reason I didn't add it here was since the cert watcher is only used in tls scenarios and I wanted to keep the tls vs non tls functionality in the method limited to `serve` vs `serverTLS`. I can add if you feel strongly about it.

Reply:
that makes sense, we can just keep it there.